File: erlang/leap/leap.erl
% Caculate leap years

-module(leap).
-export([leap_year/1]).

leap_year(Year) when Year rem 4 == 0 ->
    if 
        Year rem 100 /= 0 ->
            true;
        Year rem 400 == 0 ->
            true;
        true  ->
            false
    end;

leap_year(Year) when Year rem 4 /= 0 ->
    false.

File: csharp/leap/Leap.cs
//using System;

public class Year {

	public Year() {
	}

	public static bool IsLeap(int year) {
		if (year % 400 == 0) {
			return true;
		} else if ((year % 4 == 0) && (year % 100 != 0)) {
			return true;
		} else {
			return false;
		}
	}

}

